soccer/gameplay/evaluation/shooting.py

```diff
@@ -13,10 +13,148 @@ def eval_shot(from_point, excluded_robots=[]):
     for r in excluded_robots:
         kick_eval.add_excluded_robot(r)
     point, chance = kick_eval.eval_pt_to_opp_goal(from_point)
+    return chance
 
-    if from_point.y > (constants.Field.Length / 2):
+## Shoot through a formation of enemy robots at a target
+#
+# @param target_pos: the target to shoot at
+# @param max max_shooting_angle: The largest angle we will search to find a gap
+# @param robot_offset: How offset from an enemy robot we will shoot
+# @return a point
+def find_gap(target_pos=constants.Field.TheirGoalSegment.center(), max_shooting_angle=60, robot_offset=8, dist_from_point=.75):
 
-        return chance
+    # Find the hole in the defenders to kick at
+    # The limit is 20 cm so any point past it should be defenders right there
+    win_eval = robocup.WindowEvaluator(main.system_state())
+
+    # 500 cm min circle distance plus the robot width
+    test_distance = dist_from_point + constants.Robot.Radius
```

Comment:
We probably should expand the search radius for blocking robots a little bit more. This is a little tight